protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1456,109 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
 	return exts, nil
 }
 
+func validateHeaderType(headerType string) error {
+	// The type of the object. The value MUST be one of "string", "number", "integer", "boolean", or "array"
+	// See: https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#headerObject
+	// Note: currently not implementing array as we are only implementing this in the operation response context
+	switch headerType {
+	case
+		"string",
+		"number",
+		"integer",
+		"boolean":
+		return nil
+	}
+	return fmt.Errorf("the provided header type %q is not supported", headerType)
+}
+
+func validateDefaultValueType(headerType string, defaultValue string) error {
+	switch headerType {
+	case
+		"string":
+		if !(isQuotedString(defaultValue)) {
+			return fmt.Errorf("the provided default value %q does not match provider type %q, or is not properly quoted with escaped quotations", defaultValue, headerType)
+		}
+	case
+		"number":
+		err := isJSONNumber(strings.ToLower(defaultValue), headerType)
+		if err != nil {
+			return err
+		}
+	case
+		"integer":
+		_, err := strconv.ParseInt(defaultValue, 0, 64)
+		if err != nil {
+			return fmt.Errorf("the provided default value %q does not match provider type %q", defaultValue, headerType)
+		}
+	case
+		"boolean":
+		if !isBool(defaultValue) {
+			return fmt.Errorf("the provided default value %q does not match provider type %q", defaultValue, headerType)
+		}
+	}
+	return nil
+}
+
+func isQuotedString(s string) bool {
+	if !(strings.HasPrefix(s, "\"")) {
+		return false
+	}
+	if !(strings.HasSuffix(s, "\"")) {
+		return false
+	}
+	return true
+}
+
+func isJSONNumber(s string, t string) error {
+	// Numeric values that cannot be represented as sequences of digits (such as Infinity and NaN) are not permitted.
+	// See: https://tools.ietf.org/html/rfc4627#section-2.4
+	switch s {
+	case
+		"nan",
+		"inf",
+		"infinity":
+		return fmt.Errorf("the provided number %q is not a valid JSON number", s)
+	}
+	_, err := strconv.ParseFloat(s, 64)
+	if err != nil {
+		return fmt.Errorf("the provided default value %q does not match provider type %q", s, t)
+	}
+	return nil
+}
```

Comment:
How about this instead:
```suggestion
func isJSONNumber(s string, t string) error {
	val, err := strconv.ParseFloat(s, 64)
	if err != nil {
		return fmt.Errorf("the provided default value %q does not match provider type %q", s, t)
	}
	// Floating point values that cannot be represented as sequences of digits (such as Infinity and NaN) are not permitted.
	// See: https://tools.ietf.org/html/rfc4627#section-2.4
	if math.IsInf(val, 0) || math.IsNaN(val) {
		return fmt.Errorf("the provided number %q is not a valid JSON number", s)
	}
	
	return nil
}
```
? It's a bit nicer not to do string comparisons.